src/frequenz/sdk/actor/_power_managing/_base_classes.py

```diff
@@ -0,0 +1,189 @@
+    values reported by the microgrid API.
+
+    These bounds are adjusted to any restrictions placed by actors with higher
+    priorities.
+    """
+
+    distribution_result: power_distributing.Result | None
+    """The result of the last power distribution.
+
+    This is `None` if no power distribution has been performed yet.
+    """
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class Proposal:
+    """A proposal for a battery to be charged or discharged."""
+
+    source_id: str
+    """The source ID of the actor sending the request."""
+
+    preferred_power: Power | None
+    """The preferred power to be distributed to the batteries.
+
+    If `None`, the preferred power of higher priority actors will get precedence.
+    """
+
+    bounds: timeseries.Bounds[Power | None]
+    """The power bounds for the proposal.
+
+    These bounds will apply to actors with a lower priority, and can be overridden by
+    bounds from actors with a higher priority.  If None, the power bounds will be set to
+    the maximum power of the batteries in the pool.  This is currently and experimental
+    feature.
+    """
+
+    battery_ids: frozenset[int]
+    """The battery IDs to distribute the power to."""
+
+    priority: int
+    """The priority of the actor sending the proposal."""
+
+    request_timeout: datetime.timedelta = datetime.timedelta(seconds=5.0)
+    """The maximum amount of time to wait for the request to be fulfilled."""
+
+    include_broken_batteries: bool = False
+    """Whether to use all batteries included in the batteries set regardless the status.
+
+    If set to `True`, the power distribution algorithm will consider all batteries,
+    including the broken ones, when distributing power.  In such cases, any remaining
+    power after distributing among the available batteries will be distributed equally
+    among the unavailable (broken) batteries.  If all batteries in the set are
+    unavailable, the power will be equally distributed among all the unavailable
+    batteries in the request.
+
+    If set to `False`, the power distribution will only take into account the available
+    batteries, excluding any broken ones.
+    """
+
+    def __lt__(self, other: Proposal) -> bool:
+        """Compare two proposals by their priority.
+
+        When they have the same priority, compare them by their source ID.
+
+        Args:
+            other: The other proposal to compare to.
+
+        Returns:
+            Whether this proposal has a higher priority than the other proposal.
+        """
+        return (self.priority < other.priority) or (
+            self.priority == other.priority and self.source_id < other.source_id
+        )
+
+
+class Algorithm(enum.Enum):
+    """The available algorithms for the power manager."""
+
+    MATRYOSHKA = "matryoshka"
+
+
+class BaseAlgorithm(abc.ABC):
+    """The base class for algorithms."""
+
+    @abc.abstractmethod
+    def get_target_power(
```

Comment:
I wonder why this is called `get_target_power` although it's used to propose proposals. What about `calculate_target_power` or `determine_target_power`?

Reply:
True, `calculate_target_power` sounds way better, will rename.

Reply:
ANOTHER rename

Reply:
I am still voting for `request_..`

Reply:
It's just returning a calculated power. If I'd request something I'd suppose something happens, e.g. updating a db or switching a state .... But in this case nothing happens.
Therefore I'd find request misleading.